### PR TITLE
changing -dry-run to -dry_run

### PR DESCRIPTION
**Pull request by hallaroo**
Signed-off-by: hallaroo 

## Description
@aquarapid found a couple places in the help where [-dry_run] was replaced with [-dry-run]. This PR changes this back as [-dry-run] is not the correct usage, it should be the underscore one.

e.g.:

`SwitchReads Error: rpc error: code = Unknown desc = flag provided but not defined: -dry-run
E1116 15:14:07.388982   98844 main.go:76] remote error: rpc error: code = Unknown desc = flag provided but not defined: -dry-run
`

## Related Issue(s)
Not 100% sure when this was introduced but found the same issue in https://github.com/vitessio/vitess/pull/9177/ 

## Checklist
- [ ] Should this PR be backported?
- [ - ] Tests were added or are not required
- [ - ] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vtctl/vtctl.go [hunk 1]
```diff
@@ -217,7 +217,7 @@ var commands = []commandGroup{
 			{
 				name:   "ChangeTabletType",
 				method: commandChangeTabletType,
-				params: "[-dry-run] <tablet alias> <tablet type>",
+				params: "[-dry_run] <tablet alias> <tablet type>",
```
Comment (deepthi) [inline go/vt/vtctl/vtctl.go:220]: Should be reverted

File: go/vt/vtctl/vtctl.go [hunk 2]
```diff
@@ -522,13 +522,13 @@ var commands = []commandGroup{
 			{
 				name:   "SwitchReads",
 				method: commandSwitchReads,
-				params: "[-cells=c1,c2,...] [-reverse] -tablet_type={replica|rdonly} [-dry-run] <keyspace.workflow>",
+				params: "[-cells=c1,c2,...] [-reverse] -tablet_type={replica|rdonly} [-dry_run] <keyspace.workflow>",
```
Comment (deepthi) [inline go/vt/vtctl/vtctl.go:525]: :heavy_check_mark:

File: go/vt/vtctl/vtctl.go [hunk 3]
```diff
@@ -522,13 +522,13 @@ var commands = []commandGroup{
...
 				help:   "Switch read traffic for the specified workflow.",
 			},
 			{
 				name:   "SwitchWrites",
 				method: commandSwitchWrites,
-				params: "[-timeout=30s] [-reverse] [-reverse_replication=true] [-dry-run] <keyspace.workflow>",
+				params: "[-timeout=30s] [-reverse] [-reverse_replication=true] [-dry_run] <keyspace.workflow>",
```
Comment (deepthi) [inline go/vt/vtctl/vtctl.go:531]: :heavy_check_mark:

File: go/vt/vtctl/vtctl.go [hunk 4]
```diff
@@ -702,7 +702,7 @@ var commands = []commandGroup{
 			{
 				name:   "ApplyVSchema",
 				method: commandApplyVSchema,
-				params: "{-vschema=<vschema> || -vschema_file=<vschema file> || -sql=<sql> || -sql_file=<sql file>} [-cells=c1,c2,...] [-skip_rebuild] [-dry-run] <keyspace>",
+				params: "{-vschema=<vschema> || -vschema_file=<vschema file> || -sql=<sql> || -sql_file=<sql file>} [-cells=c1,c2,...] [-skip_rebuild] [-dry_run] <keyspace>",
```
Comment (deepthi) [inline go/vt/vtctl/vtctl.go:705]: Should be reverted

File: go/vt/vtctl/vtctl.go [hunk 5]
```diff
@@ -714,7 +714,7 @@ var commands = []commandGroup{
 			{
 				name:   "ApplyRoutingRules",
 				method: commandApplyRoutingRules,
-				params: "{-rules=<rules> || -rules_file=<rules_file>} [-cells=c1,c2,...] [-skip_rebuild] [-dry-run]",
+				params: "{-rules=<rules> || -rules_file=<rules_file>} [-cells=c1,c2,...] [-skip_rebuild] [-dry_run]",
```
Comment (deepthi) [inline go/vt/vtctl/vtctl.go:717]: Should be reverted

**Review by deepthi — CHANGES_REQUESTED** [review]
I looked through the history of the file, and only two of the changes are needed.
The others are consistent between the help and actual command syntax, so should stay the same.
It is unfortunate that the earlier commands used hyphens but the newer ones use underscores (and have already been deprecated).

**Review by ajm188 — COMMENTED** [review]
any reason we don't just switch all these to `DualFormatStringVar`? https://github.com/vitessio/vitess/blob/main/go/flagutil/flagutil.go#L140-L149

**Comment by hallaroo** [comment]
> any reason we don't just switch all these to `DualFormatStringVar`? https://github.com/vitessio/vitess/blob/main/go/flagutil/flagutil.go#L140-L149

Just to get the help changed out would you be willing to move this to be its own issue? Or let me know and I can get one added. Thanks!

**Comment by deepthi** [comment]
> any reason we don't just switch all these to `DualFormatStringVar`? https://github.com/vitessio/vitess/blob/main/go/flagutil/flagutil.go#L140-L149

No reason not to, but let's make that a separate task and not block this PR.

**Review by deepthi — APPROVED** [review]

**Pull request merged**